testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/micrometer/MicrometerRemovalTestCase.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.wildfly.test.integration.observability.micrometer;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.io.input.Tailer;
+import org.apache.commons.io.input.TailerListener;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.testcontainers.api.DockerRequired;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.as.test.shared.observability.setuptasks.AbstractSetupTask;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(Arquillian.class)
+@DockerRequired
+@RunAsClient
+public class MicrometerRemovalTestCase {
+    private static final String ERROR_MESSAGE = "Failed to publish metrics to OTLP receiver";
+    private static final Logger log = LoggerFactory.getLogger(MicrometerRemovalTestCase.class);
+    @ContainerResource
+    ManagementClient managementClient;
+
+    private String logFilePath;
+
+    @Before
+    public void getLogLocation() throws IOException {
+        // /path=jboss.server.log.dir:read-attribute(name="path")
+        ModelNode address = Operations.createAddress("path", "jboss.server.log.dir");
+        ModelNode op = Operations.createReadAttributeOperation(address, "path");
+        final ModelNode result = managementClient.getControllerClient().execute(op);
+        logFilePath = result.get("result").asString();
+    }
+
+    @Test
+    public void testRemoval() throws Exception {
+        MicrometerSetupTask setupTask = new MicrometerSetupTask();
+        ServerLogTailerListener listener = new ServerLogTailerListener();
+        try (Tailer ignored = Tailer.builder()
+            .setFile(new File(logFilePath, "server.log"))
+            .setTailerListener(listener)
+            .setDelayDuration(Duration.ofMillis(500))
+            .get()) {
+            setupTask.setup(managementClient, "");
+            Thread.sleep(1000);
```

Comment:
These sleeps should probably use `TimeoutUtil.adjust(1000)`.

Reply:
Ah, yes. I always forget about that one.